### PR TITLE
Update keyboard and syscall handlers

### PR DESCRIPTION
**Pull request by BraveJero**
Develop a cyclic buffer and update how we call the syscalls. Clean up.

### CODE REVIEW

File: Kernel/interruptions/keyboard.c [hunk 1]
```diff
@@ -8,12 +8,14 @@
...
 
 uint8_t flags = 0;
 
-static char buffer[BUFFER_SIZE];
-uint64_t tail = 0, head = 0; // head escribe, tail lee
+typedef uint8_t BufferPtr;
+
+static uint8_t buffer[BUFFER_SIZE]={0};
+static BufferPtr w_pointer = 0, r_pointer = 0;
 // {0, 0, 0, 0, 0, 0, 0, 0, 0}
```
Comment (juani-garcia) [inline Kernel/interruptions/keyboard.c:19]: This comments make no more sense and they must be deleted.

File: Kernel/interruptions/keyboard.c [hunk 2]
```diff
@@ -83,12 +76,25 @@ void keyboard_handler() {
...
+  
+  if (count > BUFFER_SIZE) {} // TODO: what do we do.
+  
+  long i = 0;
+
+  while (i < count && r_pointer != w_pointer)  // TODO: Check if we keep the last \n
+    buf[i++] = buffer[r_pointer++];
+  
+  return i;
 }
```
Comment (juani-garcia) [inline Kernel/interruptions/keyboard.c:100]: Files should end on a new line.

File: Kernel/interruptions/syscalls.h [hunk 3]
```diff
@@ -5,7 +5,8 @@
 
 #define STD_OUT 1
 #define STD_ERR 2
+#define NULL 0
```
Comment (juani-garcia) [inline Kernel/interruptions/syscalls.h:8]: NULL should be cast to (void *) to avoid warnings.

File: Kernel/interruptions/syscalls.h [hunk 4]
```diff
@@ -5,7 +5,8 @@
...
 
-uint64_t syscallDispatcher(uint64_t rdi, uint64_t rsi, uint64_t rdx, uint64_t rcx);
+uint64_t sysCallDispatcher(uint64_t rdi, uint64_t rsi, uint64_t rdx, uint64_t rax);
 
 #endif
```
Comment (juani-garcia) [inline Kernel/interruptions/syscalls.h:12]: Files should end on a new line.

File: Userland/SampleCodeModule/include/lib.h [hunk 5]
```diff
@@ -21,16 +25,18 @@ int print_f(uint8_t fd, const char * format, ...);
...
+void print(uint8_t fd, const char * s, uint64_t count);
 void setReg(void);
 uint64_t fillDate(dateType * pDate);
 void inforeg(void);
 int64_t fillMem(uint64_t src, uint8_t * dst, uint8_t size);
-int64_t getChar(void);
+long read(unsigned int fd, char * buf, size_t count);
 int _quadratic(double * a, double * b, double * c, double * root1, double * root2);
 
 #endif
```
Comment (juani-garcia) [inline Userland/SampleCodeModule/include/lib.h:42]: Files should end on new line.

**Review by juani-garcia — CHANGES_REQUESTED** [review]

File: Kernel/interruptions/keyboard.c (hunk 2 above)
Comment (BraveJero) [reply Kernel/interruptions/keyboard.c:100]: The clang-formatter should tajke care of this later

**Pull request merged**